File: ProLog.py
import tkinter as tk
from tkinter import ttk, messagebox

# =====================
# Données
# =====================
suspects = ["john", "mary", "alice", "bruno", "sophie"]
crimes = ["vol", "assassinat", "escroquerie"]

has_motive = {
    ("john", "vol"),
    ("mary", "assassinat"),
    ("alice", "escroquerie")
}

was_near_crime_scene = {
    ("john", "vol"),
    ("mary", "assassinat")
}

has_fingerprint_on_weapon = {
    ("john", "vol"),
    ("mary", "assassinat")
}

has_bank_transaction = {
    ("alice", "escroquerie"),
    ("bruno", "escroquerie")
}

owns_fake_identity = {
    ("sophie", "escroquerie")
}

# =====================
# Règles is_guilty)
# =====================
def is_guilty(suspect, crime):
    if crime == "vol":
        return (
            (suspect, "vol") in has_motive and
            (suspect, "vol") in was_near_crime_scene and
            (suspect, "vol") in has_fingerprint_on_weapon
        )

    elif crime == "assassinat":
        return (
            (suspect, "assassinat") in has_motive and
            (suspect, "assassinat") in was_near_crime_scene and
            (
                (suspect, "assassinat") in has_fingerprint_on_weapon
            )
        )

    elif crime == "escroquerie":
        return (
            ((suspect, "escroquerie") in has_motive and
             (suspect, "escroquerie") in has_bank_transaction)
            or (suspect, "escroquerie") in owns_fake_identity
        )

    return False

# =====================
# Interface Tkinter
# =====================
def check_verdict():
    suspect = suspect_var.get()
    crime = crime_var.get()

    if not suspect or not crime:
        messagebox.showwarning("Erreur", "Choisis un suspect et un crime.")
        return

    guilty = is_guilty(suspect, crime)
    if guilty:
        messagebox.showinfo("Verdict", f"{suspect.capitalize()} est COUPABLE pour {crime}.")
    else:
        messagebox.showinfo("Verdict", f"{suspect.capitalize()} est NON COUPABLE pour {crime}.")

# Fenêtre principale
root = tk.Tk()
root.title("Système Expert - Crimes")
root.geometry("400x250")

# Labels
tk.Label(root, text="Choisis un suspect :", font=("Arial", 12)).pack(pady=5)
suspect_var = tk.StringVar()
suspect_menu = ttk.Combobox(root, textvariable=suspect_var, values=suspects, state="readonly")
suspect_menu.pack()

tk.Label(root, text="Choisis un crime :", font=("Arial", 12)).pack(pady=5)
crime_var = tk.StringVar()
crime_menu = ttk.Combobox(root, textvariable=crime_var, values=crimes, state="readonly")
crime_menu.pack()

# Bouton
tk.Button(root, text="Vérifier", command=check_verdict, font=("Arial", 12), bg="lightblue").pack(pady=20)

# Lancer l'app
root.mainloop()

File: Prolog.pl
% =====================
% Types de crime
% =====================
crime_type(assassinat).
crime_type(vol).
crime_type(escroquerie).

% =====================
% Faits
% =====================
suspect(john).
suspect(mary).
suspect(alice).
suspect(bruno).
suspect(sophie).

% --- has_motive/2 ---
has_motive(john, vol).
has_motive(mary, assassinat).
has_motive(alice, escroquerie).

% --- was_near_crime_scene/2 ---
was_near_crime_scene(john, vol).
was_near_crime_scene(mary, assassinat).

% --- has_fingerprint_on_weapon/2 ---
has_fingerprint_on_weapon(john, vol).
has_fingerprint_on_weapon(mary, assassinat).

% --- has_bank_transaction/2 ---
has_bank_transaction(alice, escroquerie).
has_bank_transaction(bruno, escroquerie).

% --- owns_fake_identity/2 ---
owns_fake_identity(sophie, escroquerie).

% =====================
% Règles de culpabilité
% =====================

% --- Vol ---
is_guilty(Suspect, vol) :-
    has_motive(Suspect, vol),
    was_near_crime_scene(Suspect, vol),
    has_fingerprint_on_weapon(Suspect, vol).

% --- Assassinat ---
is_guilty(Suspect, assassinat) :-
    has_motive(Suspect, assassinat),
    was_near_crime_scene(Suspect, assassinat),
    ( has_fingerprint_on_weapon(Suspect, assassinat)
    ; eyewitness_identification(Suspect, assassinat)
    ).

% --- Escroquerie ---
is_guilty(Suspect, escroquerie) :-
    ( has_motive(Suspect, escroquerie),
      has_bank_transaction(Suspect, escroquerie)
    )
    ;
    owns_fake_identity(Suspect, escroquerie).

% =====================
% Menu interactif
% =====================
menu :-
    writeln('=============================='),
    writeln('     SYSTEME EXPERT CRIME     '),
    writeln('=============================='),
    writeln('Suspects disponibles :'),
    forall(suspect(S), (write('- '), writeln(S))),
    writeln('------------------------------'),
    writeln('Crimes possibles :'),
    forall(crime_type(C), (write('- '), writeln(C))),
    writeln('=============================='),
    writeln('Tape ton choix comme ceci:'),
    writeln('crime(nom_suspect, type_crime).'),
    writeln('Puis termine avec un point et Entrée.'),
    writeln('=============================='),
    main.

% =====================
% Entrée principale
% =====================
main :-
    current_input(Input),
    read(Input, crime(Suspect, CrimeType)),
    (   is_guilty(Suspect, CrimeType) ->
        format("Verdict : ~w est COUPABLE pour ~w.~n", [Suspect, CrimeType])
    ;   format("Verdict : ~w est NON COUPABLE pour ~w.~n", [Suspect, CrimeType])
    ),
    halt.
